nym-api/src/unstable_routes/account/data_collector.rs

```diff
@@ -171,24 +181,33 @@ impl AddressDataCollector {
 }
 
 pub(crate) struct AddressDelegationInfo {
-    delegations: Vec<nym_mixnet_contract_common::Delegation>,
-    delegated_to_nodes: HashMap<NodeId, RewardAndBondInfo>,
+    details: nym_mixnet_contract_common::Delegation,
+    node_reward_info: NodeBondStatus,
 }
 
 impl AddressDelegationInfo {
-    pub(crate) fn delegations(self) -> Vec<nym_mixnet_contract_common::Delegation> {
-        self.delegations
+    pub(crate) fn details(&self) -> &nym_mixnet_contract_common::Delegation {
+        &self.details
+    }
+
+    pub(crate) fn is_node_bonded(&self) -> bool {
+        matches!(self.node_reward_info, NodeBondStatus::Bonded(_, _))
     }
 }
 
-type RewardAndBondInfo = (NodeRewarding, bool);
+pub(crate) enum NodeBondStatus {
+    /// bool: is node in the process of unbonding
+    Bonded(NodeRewarding, bool),
```

Comment:
if you have to document what a random bool does, why not just use proper field names instead, i.e.
```rust
pub(crate) enum NodeBondStatus {
  Bonded {
    rewarding_info: NodeRewarding,
    unbonding: bool,
  },
  Unbonded,
}
```